contracts/utils/UniversalResolver.sol

```diff
@@ -7,51 +7,197 @@ import "./LowLevelCallUtils.sol";
 import "../registry/ENS.sol";
 import "../resolvers/profiles/IExtendedResolver.sol";
 import "../resolvers/Resolver.sol";
+import "./NameEncoder.sol";
 
-error OffchainLookup(address sender, string[] urls, bytes callData, bytes4 callbackFunction, bytes extraData);
+error OffchainLookup(
+    address sender,
+    string[] urls,
+    bytes callData,
+    bytes4 callbackFunction,
+    bytes extraData
+);
 
 /**
  * The Universal Resolver is a contract that handles the work of resolving a name entirely onchain,
- * making it possible to make a single smart contract call to resolve an ENS name. 
+ * making it possible to make a single smart contract call to resolve an ENS name.
  */
 contract UniversalResolver is IExtendedResolver, ERC165 {
     using Address for address;
+    using NameEncoder for string;
 
     ENS public immutable registry;
 
     constructor(address _registry) {
         registry = ENS(_registry);
     }
 
+    function encodeName(string memory name) public view returns (bytes memory) {
```

Comment:
Why is this function needed?

Reply:
I guess it's not, artifact of putting it into the library and not removing it.